Sources/SourceKitLSP/Swift/Diagnostic.swift

```diff
@@ -141,6 +143,23 @@ extension Diagnostic {
       }
     }
 
+    var code: DiagnosticCode? = nil
+    var codeDescription: CodeDescription? = nil
+    // If this diagnostic has one or more educational notes, the first one is
+    // treated as primary and used to fill in the diagnostic code and
+    // description. `useEducationalNoteAsCode` ensures a note name is only used
+    // as a code if the cline supports an extended code description.
+    if useEducationalNoteAsCode,
+       let educationalNotePaths: SKDResponseArray = diag[keys.educational_note_paths],
+       educationalNotePaths.count > 0,
+       let primaryPath = educationalNotePaths.sourcekitd.api.variant_array_get_string(educationalNotePaths.array, 0)
```

Comment:
You could add this to SKDResponseArray, e.g.

```swift
subscript(index: Int) -> String? {
  if let cstr = sourcekitd.api.variant_array_get_string(array, index) {
    return String(cString: cstr)
  }
  return nil
}
```

Reply:
good idea, done